File: app/src/main/java/com/katyabaygin/menu/DrinkDataBase.java
package com.katyabaygin.menu;

import android.app.Application;


import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.DatabaseConfiguration;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteOpenHelper;


@Database(entities = {Drink.class}, version = 1, exportSchema = false)
public abstract class DrinkDataBase extends RoomDatabase {
    public abstract DrinkDao drinkDao();

    private static final String DB_NAME = "my_drinks.db";
    private static DrinkDataBase instance = null;

    public static DrinkDataBase getInstance(Application application) {
        if (instance == null) {
            instance = Room.databaseBuilder(
                    application,
                    DrinkDataBase.class,
                    DB_NAME
            ).build();
        }
        return instance;
    }

    @NonNull
    @Override
    protected SupportSQLiteOpenHelper createOpenHelper(DatabaseConfiguration configuration) {
        return Room.databaseBuilder(configuration.context.getApplicationContext(), DrinkDataBase.class, DB_NAME).build().getOpenHelper();
    }}

File: app/src/main/java/com/katyabaygin/menu/data/DrinkDataBase.kt
package com.katyabaygin.menu.data

import android.app.Application
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.katyabaygin.menu.domain.Drink

@Database(entities = [Drink::class], version = 1, exportSchema = false)
abstract class DrinkDataBase : RoomDatabase() {
    abstract fun drinkDao(): DrinkDao

    companion object {
        private const val DB_NAME = "my_drinks.db"
        private var instance: DrinkDataBase? = null

        fun getInstance(application: Application): DrinkDataBase {
            if (instance == null) {
                synchronized(DrinkDataBase::class) {
                    if (instance == null) {
                        instance = Room.databaseBuilder(
                            application.applicationContext,
                            DrinkDataBase::class.java, DB_NAME
                        ).build()
                    }
                }
            }
            return instance as DrinkDataBase
        }
    }
}
